### ISSUE TITLE
UploadField  increase GridFieldPaginator(8))

### ISSUE THREAD

**Issue by Legin76** [silverstripe/silverstripe-framework#5007]
Please could you increase the pagination in UploadField or even better make it so that we can override it in the /mysite/_config/config.yml.

Ideally the default would be 20 to 25.
See $config->addComponent(new GridFieldPaginator(8));

Thanks

**Comment by tractorcow** [comment]
Would you like to pull request a fix?

You could add a page_size config to the `UploadField_SelectHandler` class as below.

```php

class UploadField_SelectHandler {
	private static $page_size = 11;
	
	// ...
	
	protected function getListField($folderID) {
		// ...
		$pageSize = Config::inst()->get(get_class($this), 'page_size');
		$config->addComponent(new GridFieldPaginator($pageSize));
		// ...
	}
}
```

**Comment by Legin76** [comment]
My coding isn't up to the standard to even think about doing that..

The code above appears to work but I can't work out how to add it to config.yml. 

UploadField:
    page_size: 25

**Comment by kinglozzer** [comment]
Almost @Legin76 :)

```yml
UploadField_SelectHandler:
  page_size: 25
```

There are some detailed instructions for submitting a pull request here: https://docs.silverstripe.org/en/3.2/contributing/code/. As this is a new feature, a pull request would need to be based off the `3` branch.

If those docs seem intimidating, you might even be able to make the pull request directly in Github’s UI by clicking on the “Edit” button in [UploadField.php](https://github.com/silverstripe/silverstripe-framework/blob/3/forms/UploadField.php). Good luck!

**Comment by Legin76** [comment]
Thanks for your help.Well that wasn't so bad.. 
I think I've managed to do it correctly. https://github.com/silverstripe/silverstripe-framework/pull/5011

Thanks for your help.

**Comment by kinglozzer** [comment]
@Legin76 The PR is almost perfect, except that it’s open against the wrong branch :(. As we’ve adopted semantic versioning, we can’t merge any new features into `3.1`, `3.2` or even `3.3` now.

Could I trouble you to try again, but this time when you reach the “Comparing changes” screen, pick `3` from the “base” dropdown (on the left-hand side) instead of `3.1`? If you use the following link, it should automatically be opened against that branch: https://github.com/silverstripe/silverstripe-framework/edit/3/forms/UploadField.php.

Could you also put the `private static ...` at the top of the `UploadField_SelectHandler` class with the other properties? You could add it just after line 1534 for example :)

Thanks!

**Comment by Legin76** [comment]
Attempt 2 as mentioned... 
https://github.com/silverstripe/silverstripe-framework/pull/5013

**Comment by kinglozzer** [comment]
Nice, good job @Legin76 :D. Closed via 68433cc5521a09e4877a5e5115878567471ba088
